scipy/optimize/_direct_py.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
```

Comment:
Actually, the name of the module containing FORTRAN implementation is also, `_direct` and hence, giving the same name to this file generates error like, `Unable to import _minimize_direct` (because there is not `_minimize_direct` in `_direct` module instead it is in `_direct.py`). Any suggestions to avoid this?

Reply:
rename the directory to `_direct_src`

Reply:
It's the `.so` rather than the directory name; `_directmodule` is probably a good name (IIRC that's what we do in `optimize` and `integrate` as well.